src/server/game/Entities/Player/Player.cpp

```diff
@@ -1253,7 +1253,7 @@ void Player::ToggleAFK()
     ToggleFlag(PLAYER_FLAGS, PLAYER_FLAGS_AFK);
 
     // afk player not allowed in battleground
-    if (isAFK() && InBattleground() && InArena())
+    if (isAFK() && (InBattleground() || InArena()))
```

Comment:
`InArena()` is not needed - `InBattleground()` means all battlegrounds, including arenas.